File: compatibility-tests/src/test/java/nnl/rocks/kactoos/io/DeadOutputTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.io;

import nnl.rocks.kactoos.matchers.InputHasContent;
import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.junit.Test;

/**
 * Test case for {@link DeadOutput}.
 *
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: f2a3133c21126d3f61a50b5514b74f58bcad2914 $
 * @since 0.16
 * @checkstyle JavadocMethodCheck (500 lines)
 * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
 */
public final class DeadOutputTest {

    @Test
    public void readsEmptyContent() {
        MatcherAssert.assertThat(
            "Can't write empty content",
            new TeeInput(
                new InputOf("How are you, мой друг?"),
                new DeadOutput()
            ),
            new InputHasContent(Matchers.endsWith("друг?"))
        );
    }

}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/io/DeadOutputTest.kt

package nnl.rocks.kactoos.io

import nnl.rocks.kactoos.matchers.InputHasContent
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Test

/**
 * Test case for [DeadOutput].
 *
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: f2a3133c21126d3f61a50b5514b74f58bcad2914 $
 * @since 0.16
 * @checkstyle JavadocMethodCheck (500 lines)
 * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
 */
class DeadOutputTest {

    @Test
    fun readsEmptyContent() {
        MatcherAssert.assertThat(
                "Can't write empty content",
                TeeInput(
                        InputOf("How are you, мой друг?"),
                        DeadOutput()
                ),
                InputHasContent(Matchers.endsWith("друг?"))
        )
    }
}
